test/built-ins/Promise/prototype/finally/rejected-observable-then-calls-argument.js

```diff
@@ -40,10 +40,6 @@ var expected = [
 
 var then = Promise.prototype.then;
 Promise.prototype.then = function(resolve, reject) {
-  assert.sameValue(isConstructor(reject), false, 'isConstructor(reject) must return false');
-  assert.throws(TypeError, () => {
-    new reject();
-  }, '`new reject()` throws TypeError');
```

Comment:
I don't think this was an intentional use of `isConstructor` (largely because of the misleading use of `assert.throws`), so I'm not taking it as an indication that the current behavior is desired by test authors.